contracts/token/ERC721/ERC721Basic.sol

```diff
@@ -1,80 +1,310 @@
 pragma solidity ^0.4.24;
 
-import "../../introspection/ERC165.sol";
+import "./IERC721Basic.sol";
+import "./IERC721Receiver.sol";
+import "../../math/SafeMath.sol";
+import "../../AddressUtils.sol";
+import "../../introspection/SupportsInterfaceWithLookup.sol";
 
 
 /**
- * @title ERC721 Non-Fungible Token Standard basic interface
+ * @title ERC721 Non-Fungible Token Standard basic implementation
  * @dev see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md
  */
-contract ERC721Basic is ERC165 {
-
-  bytes4 internal constant InterfaceId_ERC721 = 0x80ac58cd;
-  /*
-   * 0x80ac58cd ===
-   *   bytes4(keccak256('balanceOf(address)')) ^
-   *   bytes4(keccak256('ownerOf(uint256)')) ^
-   *   bytes4(keccak256('approve(address,uint256)')) ^
-   *   bytes4(keccak256('getApproved(uint256)')) ^
-   *   bytes4(keccak256('setApprovalForAll(address,bool)')) ^
-   *   bytes4(keccak256('isApprovedForAll(address,address)')) ^
-   *   bytes4(keccak256('transferFrom(address,address,uint256)')) ^
-   *   bytes4(keccak256('safeTransferFrom(address,address,uint256)')) ^
-   *   bytes4(keccak256('safeTransferFrom(address,address,uint256,bytes)'))
+contract ERC721Basic is SupportsInterfaceWithLookup, IERC721Basic {
+
+  using SafeMath for uint256;
+  using AddressUtils for address;
+
+  // Equals to `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+  // which can be also obtained as `ERC721Receiver(0).onERC721Received.selector`
+  bytes4 private constant ERC721_RECEIVED = 0x150b7a02;
```

Comment:
nit, but I'd prefer to name this `InterfaceId_ERC721Received` for consistency. If we're breaking backwards compat, might as well include this?

Reply:
It's a `private` constant so it doesn't affect backwards compatibility. Also, even though it happens to be an interface id, It's not it's purpose exactly. I'd rather keep that convention for ERC165 intetrfaces.